Agents/PubChemAgent/setup.py

```diff
@@ -12,18 +12,23 @@
     python_requires='>=3.5',
     include_package_data=True,
     install_requires=[
-        'numpy',
+        'numpy<2.0',
         'pyderivationagent==1.5.0',
         'docopt',
         'py4jps==1.0.36',
-        'pint',
+        'pint==0.21.1',
+        'flexparser',
+        'flexcache',
         'rdkit',  
         'Flask==2.1.0',  
-        'Werkzeug==2.3.6',  
+        'Werkzeug>=2.1,<2.2',  
         'periodictable',
         'bioservices',
         'unit-parse'  # Verify the correct package name on PyPI
     ],
```

Comment:
I guess you will need to update some of the versions here, once you switch to a more recent python base image.